pkg/process/runner/submitter.go

```diff
@@ -75,22 +75,22 @@ type CheckSubmitter struct {
 	rtNotifierChan chan types.RTResponse
 }
 
-func NewSubmitter(cfg ddconfig.ConfigReader, hostname string) (*CheckSubmitter, error) {
-	queueBytes := cfg.GetInt("process_config.process_queue_bytes")
+func NewSubmitter(config config.Component, hostname string) (*CheckSubmitter, error) {
```

Comment:
Nit: I think we don't want to do this. Ideally for us, library code should be able to be used without fx.